framework/include/parser/CommandLine.h

```diff
@@ -172,40 +175,75 @@ CommandLine::setArgument(std::stringstream & stream, T & argument)
   stream >> argument;
 }
 
+template <typename T>
+void
+CommandLine::setArgument(std::stringstream & stream, T & argument, const std::string & cli_switch)
+{
+  // Keep track of and change the throw on error characteristics so that
+  // we can catch parsing errors for the argument
+  const auto throw_on_error_orig = Moose::_throw_on_error;
+  Moose::_throw_on_error = true;
+
+  const auto raw_value = stream.str();
+  try
+  {
+    setArgument(stream, argument);
+  }
+  catch (std::exception & e)
+  {
+    Moose::_throw_on_error = throw_on_error_orig;
+    mooseError("While parsing command line argument '",
+               cli_switch,
+               "' with value '",
+               raw_value,
+               "':\n\n",
+               e.what());
```

Comment:
Do you have an example of what this error message looks like?

Reply:
I meant to test this, just forgot it. But it's important here for catching the error when you provide a bad enum.

Before this:

```
***ERROR***

Invalid option "baz" in MooseEnum.  Valid options (not case-sensitive) are "foo bar"." thrown in the test body.
```

After this:
```
***ERROR***

While parsing command line argument '--value' with value 'baz':

Invalid option "baz" in MooseEnum.  Valid options (not case-sensitive) are "foo bar"." thrown in the test body.
```